gui/dig.lua

```diff
@@ -683,29 +798,80 @@ function GenericOptionsPanel:init()
+            visible = function() return self.dig_panel.subviews.mode_name:getOptionValue().desig == "i" end,
             options = stair_options,
         },
         widgets.CycleHotkeyLabel {
             view_id = "stairs_bottom_subtype",
-            key = "CUSTOM_B",
+            key = "CUSTOM_N",
             label = "Bottom Stair Type: ",
+            frame = { l = 1 },
             active = true,
             enabled = true,
-            visible = function() return self.dig_panel.subviews.mode_name:getOptionValue() == "i" end,
+            visible = function() return self.dig_panel.subviews.mode_name:getOptionValue().desig == "i" end,
             options = stair_options,
         },
+        widgets.ResizingPanel {
+            view_id = 'transform_panel_rotate',
+            visible = function() return self.dig_panel.subviews.mode_name:getOptionValue().desig == "b" end,
+            subviews = {
+                widgets.Label {
+                    view_id = "building_outer_config",
+                    frame = { t = 0, l = 1 },
+                    text = { { tile = BUTTON_PEN_LEFT }, { tile = HELP_PEN_CENTER }, { tile = BUTTON_PEN_RIGHT } },
+                    on_click = function()
+                        view.help_window.message = CONSTRUCTION_HELP
+                        view.help_window.visible = true
+                        view:updateLayout()
+                    end
+                },
+                widgets.CycleHotkeyLabel {
+                    view_id = "building_outer_tiles",
+                    key = "CUSTOM_R",
+                    label = "Outer Tiles: ",
+                    frame = { t = 0, l = 5 },
+                    active = true,
+                    enabled = true,
+                    initial_option = 1,
+                    visible = function() return self.dig_panel.subviews.mode_name:getOptionValue().desig == "b" end,
+                    options = build_options,
+                },
+                widgets.Label {
+                    view_id = "building_inner_config",
+                    frame = { t = 1, l = 1 },
+                    text = { { tile = BUTTON_PEN_LEFT }, { tile = HELP_PEN_CENTER }, { tile = BUTTON_PEN_RIGHT } },
+                    on_click = function()
+                        view.help_window.message = CONSTRUCTION_HELP
+                        view.help_window.visible = true
+                        view:updateLayout()
+                    end
+                },
+                widgets.CycleHotkeyLabel {
+                    view_id = "building_inner_tiles",
+                    key = "CUSTOM_G",
+                    label = "Inner Tiles: ",
+                    frame = { t = 1, l = 5 },
+                    active = true,
+                    enabled = true,
+                    initial_option = 2,
+                    visible = function() return self.dig_panel.subviews.mode_name:getOptionValue().desig == "b" end,
```

Comment:
in order to avoid copying this function everywhere it's needed, you can declare a local function inside the scope if `init` and refer to it by name. e.g. https://github.com/DFHack/dfhack/blob/develop/plugins/lua/buildingplan/planneroverlay.lua#L156

Reply:
similar suggestion for the on_click callbacks above